src/interpreter.c

```diff
@@ -144,7 +141,11 @@ static void eval_abstracttype(jl_expr_t *ex, interpreter_state *s)
         name = (jl_value_t*)jl_globalref_name(name);
     }
     assert(jl_is_symbol(name));
-    dt = jl_new_abstracttype(name, modu, NULL, (jl_svec_t*)para);
+    if (jl_is_globalref(args[2])) {
```

Comment:
I think this is a little too unpredictable, since it can change e.g. based on whether the definition came from a macro.